packages/itwin/grouping-mapping-widget/src/widget/GroupingMappingWidget.tsx

```diff
@@ -14,19 +14,19 @@ import {
 } from "@itwin/appui-abstract";
 import type { AccessToken } from "@itwin/core-bentley";
 import React from "react";
-import GroupingMapping from "./components/GroupingMapping";
-import type { ClientPrefix } from "./components/GroupingMapping";
+import type { ClientPrefix } from "./components/context/GroupingApiConfigContext";
+import GroupingMapping, { GroupingMappingProps } from "./components/GroupingMapping";
 import type { IMappingClient } from "./IMappingClient";
 
 export class GroupingMappingProvider implements UiItemsProvider {
   public readonly id = "GroupingMappingProvider";
 
-  private readonly _accessToken?: AccessToken;
+  private readonly _getAccessToken?: () => Promise<AccessToken>;
   private readonly _prefix?: ClientPrefix;
   private readonly _client?: IMappingClient;
 
-  constructor(accessToken?: AccessToken, prefix?: "" | "dev" | "qa", client?: IMappingClient) {
-    this._accessToken = accessToken;
+  constructor({ getAccessToken, prefix, client }: GroupingMappingProps = {}) {
```

Comment:
I don't think it's worth to deconstruct props here since later the same props object is rebuilt. If this was not reconstructed then it could be used like this: `<GroupingMapping {...this._props} />`.

Reply:
I agree. Changed it to pass the object directly.

Reply:
It could be simplified even further - the separate field declaration is not needed, there is a shorthand for constructors:
`constructor(private readonly _props: GroupingMappingProps = {}) {}`

Reply:
Oh right completely forgot about that shorthand. Thanks!